### ISSUE TITLE
Torchscript Variable Length Tuples

### ISSUE THREAD

**Issue by randolf-scholz** [pytorch/pytorch#64700]
## 🚀 Feature

It would be nice if torchscript would support variable length tuples like `tuple[int, ...]`. Consider the following example:

```python
import torch
from typing import Optional, Union

def scaled_norm(
    x: torch.Tensor,
    dim: Optional[Union[int, tuple[int, ...]]] = None,
    p: float = 2.0,
    keepdim: bool = False,
) -> torch.Tensor: 
    return torch.mean(x**p, dim=dim, keepdim=keepdim) ** (1/p)
```

This is a commonly used variant of the p-norm which is averaged against the number of elements. Currently, it is impossible to torch-script this function in a way that is both compatible with `torchscript` and `mypy`, the former requiring to change the `Optional[Union[int, tuple[int, ...]]]` into a `list[int]`.

**Comment by randolf-scholz** [comment]
To add to this, type hinting with `list[int]` is problematic because then we have to set `axis: list[int] = []`, i.e. have a mutable default argument which is a big no-no.

**Comment by randolf-scholz** [comment]
jit is deprecated
